Stats: Use yearly product tier pricing as default and a refactoring

kangzj opened the pull request:
## Proposed Changes

* Use yearly product tier pricing as default
* Use `maximum_units` for display
* Refactored the tiers logic to a hook like it was, which I think looks a bit clearer and we don't have to pass down the tiers for so many levels

## Testing Instructions

Please follow instructions for https://github.com/Automattic/wp-calypso/pull/84535, but note the pricing is based on yearly and the units are maximum.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

kangzj committed:
refactor how to get and transform tiers

kangzj commented on client/my-sites/stats/stats-purchase/use-available-upgrade-tiers.tsx:
```diff
@@ -0,0 +1,85 @@
+function useAvailableUpgradeTiers( siteId: number | null ): StatsPlanTierUI[] {
+	// 1. Get the tiers.
+	const commercialProduct = useSelector( ( state ) =>
+		getProductBySlug( state, PRODUCT_JETPACK_STATS_YEARLY )
```
use yearly product as default

kangzj commented on client/my-sites/stats/stats-purchase/use-available-upgrade-tiers.tsx:
```diff
@@ -0,0 +1,85 @@
+		// highest tier extension
+		return {
+			price: tier?.minimum_price_monthly_display,
+			views: tier?.maximum_units,
```
use `maximum_units` for display

kangzj commented on client/my-sites/stats/stats-purchase/use-available-upgrade-tiers.tsx:
```diff
@@ -0,0 +1,85 @@
+	{
+		price: '$9',
+		views: 10000,
+		description: '$9/month for 10k views',
```
We don't translate mock data 😄

a8ck3n committed:
Comments++

a8ck3n reviewed (approved):
Works beautifully!

Just added some inline comments on the data hook based on your own comments. Will merge this once the tests pass again.

a8ck3n committed:
Small fix for type error.

This isn’t really failing gracefully though. It probably makes sense to have a check for validity as part of the data pull. That way we can disable the slider if we get unexpected data from the API.

dognose24 commented on client/my-sites/stats/stats-purchase/stats-purchase-tier-upgrade-slider.tsx:
```diff
@@ -39,99 +32,53 @@ function useTranslatedStrings() {
 					<h2>{ translatedStrings.limits }</h2>
 					<p className="left-aligned">
-						{ plans[ currentPlanIndex ]?.extension ? (
+						{ tiers[ currentPlanIndex ]?.extension ? (
```
Could we reuse the flag `hasExtension` here?

kangzj replied:
addressed in c8c8dc3a72

dognose24 commented on client/my-sites/stats/stats-purchase/stats-purchase-tier-upgrade-slider.tsx:
```diff
@@ -39,99 +32,53 @@ function useTranslatedStrings() {
 							<>
 								<span>+</span>
-								<ShortenedNumber value={ EXTENSION_THRESHOLD * 1000000 } />
+								<ShortenedNumber value={ lhValue } />
```
Maybe the `lhValue` is unnecessary to check again under this conditioned DOM rendering. For example, simultaneously determine the rendering in the `lhValue` check above.

kangzj replied:
Addressed in 75ff25cda4

dognose24 commented on client/my-sites/stats/stats-purchase/stats-purchase-tier-upgrade-slider.tsx:
```diff
@@ -147,18 +94,18 @@ function TierUpgradeSlider( { className, priceTiers, currencyCode }: TierUpgrade
 				position="right"
 				context={ infoReferenceElement?.current }
-				isVisible={ plans[ currentPlanIndex ]?.extension }
+				isVisible={ tiers[ currentPlanIndex ]?.extension }
```
Could we reuse the flag `hasExtension` here again?

kangzj replied:
Addressed c8c8dc3a72

dognose24 reviewed (commented):
Work as described! 👍🏼 Left some suggestions for probably making the code cleaner.

kangzj committed:
reuse hasExtension

kangzj committed:
reduce dup code

kangzj committed:
+ should be in the right hand side mostly

kangzj committed:
divide yearly unit price by 12

dognose24 commented on client/my-sites/stats/stats-purchase/stats-purchase-tier-upgrade-slider.tsx:
```diff
@@ -54,7 +54,7 @@ function TierUpgradeSlider( { className, currencyCode }: TierUpgradeSliderProps
 	// Should consider validating the inputs before displaying them.
 	// The following will draw a "-" for the views value if it's undefined.
-	const hasExtension = tiers[ currentPlanIndex ]?.extension || false;
+	const hasExtension = !! tiers[ currentPlanIndex ]?.extension;
```
👍🏼

kangzj replied:
@a8ck3n this is a ideological pattern where we want boolean 😄
